aten/src/ATen/native/native_functions.yaml

```diff
@@ -6443,6 +6446,16 @@
     SparseCsrCUDA: sparse_sampled_addmm_sparse_csr_cuda
     SparseCsrCPU: sparse_sampled_addmm_sparse_csr_cpu
 
+- func: _sparse_mm_reduce_impl(Tensor self, Tensor other, str reduce, *, Tensor? row_indices=None, Tensor? ccol_indices=None, Tensor? csr2csc=None) -> (Tensor, Tensor)
+  python_module: sparse
+  dispatch:
+    SparseCsrCPU: _sparse_mm_reduce_impl_sparse_csr_cpu
```

Comment:
Why not also register a kernel for CSC?

Reply:
Shall we put this in the next step ? Also maybe add COO, though COO format is difficult to be directly optimized from performance perspective.

Reply:
But you already have support for CSC via the `csr2csc` flag as far as I understand.

Reply:
@mingfeima, does `csr2csc` determine the output layout of the operation?

The documentation says 
```
csr2csc (Tensor, optional): permute pattern to convert :attr:`input` from CSR to CSC.
```

We have built in support for this via transpose. Try creating a 2D CSR Tensor via `to_sparse_csr` and apply `transpose(0, 1)`. It should return a CSC Tensor and not do any operations.

Reply:
Oh cool, i thought i need to rewrite the whole set of kernels for CSC. Thanks for the pointer!

Reply:
@cpuhrsch could you please help review this again ? thanks!